web/src/worker/client.ts

```diff
@@ -0,0 +1,67 @@
+import { WorkerMessageType } from './events';
+import { postMessage } from './utils';
+import { VideoReader } from '../core/video-reader';
+import { WorkerPAGFile } from './pag-file';
+import { WorkerPAGView } from './pag-view';
+import { WorkerPAGImage } from './pag-image';
+
+import type { WorkerMessage } from './worker';
+import type { ModuleOption } from '../pag';
+import type { TimeRange } from '../interfaces';
+
+export const MAX_ACTIVE_WORKER_CONTEXTS = 4;
+
+const videoReaders: VideoReader[] = [];
+
+export const createPAGWorker = (
+  moduleOption: ModuleOption & { workerScriptURL?: string | URL; workerOptions?: WorkerOptions } = {},
+) => {
+  const scriptURL = moduleOption.workerScriptURL || 'libpag.js';
+  const worker = new Worker(scriptURL, moduleOption.workerOptions);
```

Comment:
Creation of Worker may fail if scriptURL is wrong, we can throw a custom message to show how to place the libpag.js worker script